### PR TITLE
Added nonce feature to msal node

### PR DESCRIPTION
**Pull request by mpminayo**

### CODE REVIEW

File: change/@azure-msal-node-694500a8-c2bd-4902-a5ed-4d4b206e0c77.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "added nonce feature ",
```
Comment (jasonnutter) [inline change/@azure-msal-node-694500a8-c2bd-4902-a5ed-4d4b206e0c77.json]:
```suggestion
  "comment": "Added nonce support to MSAL Node #4504",
```

**Review by jasonnutter — APPROVED** [review]
Looks good, thanks for the tests and sample!

File: samples/msal-node-samples/auth-code/index.js [hunk 2]
```diff
@@ -29,21 +31,51 @@ const cachePlugin = require('../cachePlugin')(cacheLocation);
...
         const { authCodeUrlParameters } = requestConfig;
+
+        const cryptoProvider = new msal.CryptoProvider()
             
         if (req.query) {
             // Check for the state parameter
             if(req.query.state) authCodeUrlParameters.state = req.query.state;
+            // Check for nonce parameter
+            /**
+             * MSAL node provides for the use of OIDC nonce feature which is used to protect against token replay.
```
Comment (hectormmg) [inline samples/msal-node-samples/auth-code/index.js]: I'm nitpicking here, just a small suggestion
```suggestion
             * MSAL Node supports the OIDC nonce feature which is used to protect against token replay.
```

File: samples/msal-node-samples/auth-code/index.js [hunk 3]
```diff
@@ -72,18 +106,21 @@ const getTokenAuthCode = function (scenarioConfig, clientApplication, port) {
 
     app.get("/redirect", (req, res) => {
         const tokenRequest = { ...requestConfig.tokenRequest, code: req.query.code };
+        const authCodeResponse = {nonce :req.session.nonce, code: req.query.code}
```
Comment (hectormmg) [inline samples/msal-node-samples/auth-code/index.js]: Nitpick:
```suggestion
        const authCodeResponse = { nonce: req.session.nonce, code: req.query.code }
```

**Review by hectormmg — APPROVED** [review]
Looks great! A couple comments/suggestions but nothing blocking merge in my opinion.

**Review by bmahall — APPROVED** [review]

**Review by samuelkubai — APPROVED** [review]
Other than the comments from Jason and Hector, looks good to me.

**Pull request merged**